vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt

```diff
@@ -598,9 +600,16 @@ class TimelineViewModel @AssistedInject constructor(
         }
     }
 
-    private fun handleStartVoiceBroadcast() {
-        // Todo implement start voice broadcast action
-        Timber.d("Start voice broadcast clicked")
+    private fun handleVoiceBroadcastAction(action: RoomDetailAction.VoiceBroadcastAction) {
+        if (room == null) return
+        viewModelScope.launch {
+            when (action) {
+                RoomDetailAction.VoiceBroadcastAction.Pause -> voiceBroadcastHelper.pauseVoiceBroadcast(room.roomId)
+                RoomDetailAction.VoiceBroadcastAction.Resume -> voiceBroadcastHelper.resumeVoiceBroadcast(room.roomId)
+                RoomDetailAction.VoiceBroadcastAction.Start -> voiceBroadcastHelper.startVoiceBroadcast(room.roomId)
+                RoomDetailAction.VoiceBroadcastAction.Stop -> voiceBroadcastHelper.stopVoiceBroadcast(room.roomId)
+            }
```

Comment:
Nit: I would keep the same logical order for the when cases: Start/Pause/Resume/Stop, as in the declaration of `RoomDetailAction.VoiceBroadcastAction`

Reply:
https://github.com/vector-im/element-android/pull/7273/commits/41aa2c9c8d93982d0cc1d01a16207c47c322612b